cowprotocol/services
# Store executed surplus fee

Fixes #1403 

In order to correctly compute the solver rewards for settling partially fillable limit orders we need to store the fee they computed in our DB.
All of the required logic was already in place. We just had to use the `solver_fee` as the `surplus_fee` for partially fillable limit orders in the `Trade` logic.

### Test Plan
Manual test:
* temporarily adjusted test logic to not clear the DB after a test
* ensured the test solver + autopilot are correctly authenticated (to actually store the competition info)
* ran `cargo test -- --ignored partially_fillable` (e2e test that settles a partially fillable order)
* verified that we have an entry in the `order_execution` table with the expected `surplus_fee` amount in the docker testing DB

Extended one existing e2e test to verify that `executed_X_amount`s get updated in the metadata after a partial fill
